recipes/gmp/all/conanfile.py

```diff
@@ -12,22 +12,27 @@ class GmpConan(ConanFile):
     license = ("LGPL-3.0", "GPL-2.0")
     homepage = "https://gmplib.org"
     settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False], "fPIC": [True, False], "disable_assembly": [True, False],
+    options = {"shared": [True, False], "fPIC": [True, False],
+               "disable_assembly": [True, False], "enable_fat": [True, False],
                "run_checks": [True, False], "enable_cxx" : [True, False]}
-    default_options = {"shared": False, "fPIC": True, "disable_assembly": True, "run_checks": False, "enable_cxx" : True}
+    default_options = {"shared": False, "fPIC": True, "disable_assembly": True, "enable_fat": False, "run_checks": False, "enable_cxx" : True}
 
     _source_subfolder = "source_subfolder"
     _autotools = None
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if self.settings.arch not in ["x86", "x86_64"]:
+            del self.options.enable_fat
 
     def configure(self):
         if self.settings.compiler == "Visual Studio":
             raise ConanInvalidConfiguration("The gmp package cannot be built on Visual Studio.")
         if self.options.shared:
             del self.options.fPIC
+        if self.options.enable_fat:
+            del self.options.disable_assembly
```

Comment:
It seems these two settings are at odds! I am not sure there's minimal way to clean this up...

deprecate the old one + add a new one ` assembly = [ False, True, "fat"] ?just a thought!

Reply:
That's a good idea and would have been a better presentation of the available options, but GMP provides two separate flags and errors if you disable assembly and enable fat. I thought it was more important to stick to the interface that GMP provides than to improve it (it makes finding the options easier). Though I don't mind changing it if you think this'll be clearer.

Reply:
No I think what you did is more than acceptable